### ISSUE TITLE
[BUG] False Positive for `HaveOccurred()` matcher with `.Error()`

### ISSUE THREAD

**Issue by maboehm** [nunnatsa/ginkgolinter#173]
**Describe the bug**
When using the `.Error()` chaining method on an expectation, the linter incorrectly finds an issue, even though this exact pattern is shown in the official documentation as 

```go
Ω(MultipleReturnValuesFunc()).Error().ShouldNot(HaveOccurred())
```

See https://onsi.github.io/gomega/#handling-errors

**To Reproduce**
```go
package main_test

import (
	"testing"

	. "github.com/onsi/ginkgo/v2"
	. "github.com/onsi/gomega"
)

func TestE2E(t *testing.T) {
	RegisterFailHandler(Fail)
	RunSpecs(t, "demo")
}

func foo() (bool, error) {
	return false, nil
}

var _ = Describe("demo", func() {
	It("should work", func() {
		Expect(foo()).Error().NotTo(HaveOccurred())
	})
})
```

Returns
```
main_test.go:21:3: ginkgo-linter: asserting a non-error type with HaveOccurred matcher
```

**Expected behavior**
Get no warning.

**Environment:**
 - OS: macos
 - Version latest (0.18.1)
 - go: 1.22.9

**Additional context**
We first observed this when bumping golangci-lint, which bumped this linter from 0.16.0 to 0.18.0

**Comment by nunnatsa** [comment]
Just release [v0.18.2](https://github.com/nunnatsa/ginkgolinter/releases/tag/v0.18.2) with a fix to this bug.
